src/Pos/Worker.hs

```diff
@@ -58,6 +60,12 @@ allWorkers NodeResources {..} = mconcatPair
     , wrap' "delegation" $ dlgWorkers
     , wrap' "slotting"   $ (properSlottingWorkers, mempty)
 
+    , case ncTopology ncNetworkConfig of
+        TopologyBehindNAT dnsDomains ->
+          wrap' "subscription" $ subscriptionWorkers ncNetworkConfig dnsDomains
+        _otherwise ->
+          mempty -- No subscription worker required
```

Comment:
Oh, wait, I see that Alex already expressed that dependency elsewhere. I've done the same. I've introduced `topologyRunKademlia` with definition

```
topologyRunKademlia :: Topology -> Bool
topologyRunKademlia = go
  where
    go (TopologyStatic _ _)    = True
    go (TopologyBehindNAT _)   = False
    go (TopologyP2P)           = True
    go (TopologyTraditional)   = True
    go (TopologyLightWallet _) = False
```

Not _entirely_ sure if this definition is correct; would be good if @dcoutts or @avieth could verify.